pkg/util/ansible_execution.go

```diff
@@ -30,80 +36,150 @@ import (
-		Name:      "inventory",
-		MountPath: "/runner/inventory/hosts",
-		SubPath:   "inventory",
-	}
+		}
+		inventoryMount := corev1.VolumeMount{
+			Name:      "inventory",
+			MountPath: "/runner/inventory/hosts",
+			SubPath:   "inventory",
+		}
+
+		ansibleEEMounts.Volumes = append(ansibleEEMounts.Volumes, sshKeyVolume)
+		ansibleEEMounts.Volumes = append(ansibleEEMounts.Volumes, inventoryVolume)
+		ansibleEEMounts.Mounts = append(ansibleEEMounts.Mounts, sshKeyMount)
+		ansibleEEMounts.Mounts = append(ansibleEEMounts.Mounts, inventoryMount)
+
+		ansibleEE.Spec.ExtraMounts = []storage.VolMounts{ansibleEEMounts}
+
+		err := controllerutil.SetControllerReference(instance, ansibleEE, helper.GetScheme())
+		if err != nil {
+			return err
+		}
 
-	ansibleEEMounts.Volumes = append(ansibleEEMounts.Volumes, sshKeyVolume)
-	ansibleEEMounts.Volumes = append(ansibleEEMounts.Volumes, inventoryVolume)
-	ansibleEEMounts.Mounts = append(ansibleEEMounts.Mounts, sshKeyMount)
-	ansibleEEMounts.Mounts = append(ansibleEEMounts.Mounts, inventoryMount)
+		return nil
 
-	ansibleEE.Spec.ExtraMounts = append(ansibleEE.Spec.ExtraMounts, ansibleEEMounts)
+	})
 
-	err := helper.GetClient().Create(ctx, ansibleEE)
 	if err != nil {
 		util.LogErrorForObject(helper, err, fmt.Sprintf("Unable to create AnsibleEE %s", ansibleEE.Name), ansibleEE)
 		return err
 	}
 
 	return nil
 }
+
+// GetAnsibleExecution gets and returns an OpenStackAnsibleEE with the given label
+func GetAnsibleExecution(instance client.Object, ctx context.Context, helper *helper.Helper, label string) (*ansibleeev1alpha1.OpenStackAnsibleEE, error) {
+
+	var err error
+	ansibleEEs := &ansibleeev1alpha1.OpenStackAnsibleEEList{}
+
+	listOpts := []client.ListOption{
+		client.InNamespace(instance.GetNamespace()),
+	}
+	labelSelector := map[string]string{
+		label: instance.GetName(),
+	}
+	if len(labelSelector) > 0 {
+		labels := client.MatchingLabels(labelSelector)
+		listOpts = append(listOpts, labels)
+	}
+	err = helper.GetClient().List(ctx, ansibleEEs, listOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	var ansibleEE *ansibleeev1alpha1.OpenStackAnsibleEE
+	if len(ansibleEEs.Items) == 0 {
+		executionName := names.SimpleNameGenerator.GenerateName(label)
+		ansibleEE = &ansibleeev1alpha1.OpenStackAnsibleEE{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      executionName,
+				Namespace: instance.GetNamespace(),
+				Labels: map[string]string{
+					label: instance.GetName(),
+				},
+			},
+		}
+	} else if len(ansibleEEs.Items) == 1 {
+		ansibleEE = &ansibleEEs.Items[0]
+	} else {
+		return nil, errors.New(fmt.Sprintf("Multiple OpenStackAnsibleEE's found with label %s", label))
+	}
+
+	return ansibleEE, nil
+
+}
+
+// GetAnsibleExecutionJob
+func GetAnsibleExecutionJob(instance client.Object, ctx context.Context, helper *helper.Helper, label string) (*batchv1.Job, error) {
+	ansibleEE, err := GetAnsibleExecution(instance, ctx, helper, label)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
need a wait here as there is no guarantee the Job is already created.

Reply:
now you can just watch the conditions, right?